### ISSUE TITLE
services.emacs.package does not add packages from main package set

### ISSUE THREAD

**Issue by doyougnu** [NixOS/nixpkgs#96975]
**Describe the bug**
This may just be confusing documentation.

I'm trying to enable the emacs system service, in `configuration.nix` as the root user with a custom set of packages. 

The nixos manual states:

```txt
 The services.emacs.package option allows a custom derivation to be used, for example, one created by emacsWithPackages. 
```

so I have this (minimum example):

```
...
  # Set emacs to daemon mode
  services.emacs.enable = true;
  services.emacs.package = pkgs.emacsWithPackages (epkgs: (with epkgs.melpaStablePackages; [
      # From main packages set
    pkgs.notmuch
  ]));
...
```

But if I use `emacsclient --create-frame`, open `eshell` and try to find `notmuch` the package is not found. The package *is* found if I *do not* connect to the client and run `emacs` from the command line.

**To Reproduce**
1. In `configuration.nix` add the code snippet above
2. run:
```
sudo nixos-rebuild switch
systemctl --user daemon-reload
systemctl --user restart emacs.service
```
3. Connect to the daemon with `emacsclient --create-frame` and try to find `notmuch` in any way. I have been running `eshell` then looking for `notmuch`

**Expected behavior**
Emacs daemon behavior is identical to standalone emacs behavior, i.e., `notmuch` is found in daemon mode _and_ in standalone mode

**Additional context**
I have not tested melpa or elpa packages

**Notify maintainers**
@lovek323 @peti @the-kenny @jwiegley

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.61, NixOS, 20.03.2868.ff6a070b4ef (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"nixos-20.03.2868.ff6a070b4ef"`
 - channels(doyougnu): `"u-20.09pre239318.c59ea8b8a0e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by kamidon** [comment]
I think the problem here is that the elisp is in a separate output. Try:
```
services.emacs.package = pkgs.emacsWithPackages (epkgs: [
    pkgs.notmuch.emacs
  ] ++ (with epkgs.melpaStablePackages; [
      # Melpa packages you want.
  ]));
```

**Comment by doyougnu** [comment]
Thanks for the ping. Closing this, as the issue was resolved once I upgraded to 21.05.
